src/radembedded_async.c

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (C) 2022 Cadami GmbH info@cadami.net
+ *
+ * radembedded.c - a sample c program showing how to embed the configuration of a radius
+ * client, using the FreeRADIUS Client Library without an external configuration file.
+ */
+
+#include <stdlib.h>
+#include <sys/types.h>
+#include <syslog.h>
+#include <radcli/radcli.h>
+#include <string.h>
+
+#include <poll.h>
+
+#define AUTH_PORT_PASSWORD ":1812:testing123"
+#define ACCT_PORT_PASSWORD ":1813:testing123"
+#define SERVER_ADDR "localhost"
+
+int
+main (int argc, char **argv)
+{
+	int i = 0;
+	rc_handle	*rh = NULL;
+	uint32_t	client_port = 0;
+	uint32_t	status_type = PW_STATUS_STOP;
+	VALUE_PAIR	*send = NULL;
+	rc_socket_type  so_type = RC_SOCKET_UDP;
+
+	char		username[255] = "bob@somedomain.here";
+	char		callfrom[255] = "8475551212";
+	char		callto[255] = "8479630116";
+	char		myuuid[255] = "981743-asdf-90834klj234";
+	char		auth_server_ip[255] = {0};
+	char		acct_server_ip[255] = {0};
+	char		*server_ip = NULL;
+
+	int count = 10;
+	struct rc_async_handle *req = NULL;
+
+	if (argc > 1)
+		server_ip = argv[1];
+
+	if (argc > 2) {
+		if (!strcasecmp("tcp", argv[2])) {
+			so_type = RC_SOCKET_TCP;
+		} else if (!strcasecmp("udp", argv[2])) {
+			so_type = RC_SOCKET_UDP;
+		} else {
+			printf("ERROR: Invalid protocol: %s\nUse tcp or udp.\n",
+				argv[2]);
+			exit(EXIT_FAILURE);
+		}
+	}
+
+	if (argc > 3) {
+		count = atoi(argv[3]);
+		if (count < 1) {
+			printf("ERROR: Invalid request count: %s\n", argv[3]);
+			exit(EXIT_FAILURE);
+		}
+	}
+
+	if (argc > 4) {
+		printf("ERROR: Invalid number of arguments.\n");
+		exit(EXIT_FAILURE);
+	}
+
+/* ======================== Prepare New Request ============================= */
```

Comment:
It seems the function is too long. Should we split into different functions for easier readbility?